android/java/org/chromium/chrome/browser/crypto_wallet/util/TransactionUtils.java

```diff
@@ -0,0 +1,50 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.crypto_wallet.util;
+
+import org.chromium.brave_wallet.mojom.CoinType;
+import org.chromium.brave_wallet.mojom.FilTxData;
+import org.chromium.brave_wallet.mojom.SolanaTxData;
+import org.chromium.brave_wallet.mojom.TxData1559;
+import org.chromium.brave_wallet.mojom.TxDataUnion;
+
+public class TransactionUtils {
+    public static @CoinType.EnumType int getCoinFromTxDataUnion(TxDataUnion txDataUnion) {
+        if (txDataUnion.which() == TxDataUnion.Tag.FilTxData)
+            return CoinType.FIL;
+        else if (txDataUnion.which() == TxDataUnion.Tag.SolanaTxData)
+            return CoinType.SOL;
+        else
+            return CoinType.ETH;
+    }
+
+    public static TxData1559 safeEthTxData1559(TxDataUnion txDataUnion) {
+        try {
+            return txDataUnion.getEthTxData1559();
+        } catch (AssertionError e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    public static SolanaTxData safeSolTxData1559(TxDataUnion txDataUnion) {
+        try {
+            return txDataUnion.getSolanaTxData();
+        } catch (AssertionError e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    public static FilTxData safeFilTxData1559(TxDataUnion txDataUnion) {
+        try {
+            return txDataUnion.getFilTxData();
+        } catch (AssertionError e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
```

Comment:
All `safe...` methods are incorrect. They won't work as expected in Release mode as there won't be any `AssertionError` as asserts are ignored in Release mode. the safe methods should be replaced to something like that
```
    public static FilTxData safeFilTxData1559(TxDataUnion txDataUnion) {
        if (txDataUnion.which() == TxDataUnion.Tag.FilTxData) {
               return txDataUnion.getFilTxData();
        }

        return null;
    }
```

Reply:
I know that release not gonna be affected but as mentioned in slack, I need to make these thing safe proof for debug mode to implement Solana send.
Since it's being used in the [ParsedTransaction](https://github.com/brave/brave-core/blob/cdf2a96e803e4656370ce415fde14d993c6aff5d/android/java/org/chromium/chrome/browser/crypto_wallet/util/ParsedTransaction.java#L249) class so I needed a workaround. No point in updating the existing logic [which also needs to be updated again](https://github.com/brave/brave-browser/issues/24332).

Reply:
we cannot merge it like that with a potentially wrong solution. Update `safe...` functions to have the consistent behaviour between Release and Debug modes. That issue or will reuse the `safe...` functions or will remove them https://github.com/brave/brave-browser/issues/24332

Reply:
I did updated the [dev concern](https://github.com/brave/brave-browser/issues/24332) though will update it in the next push